doc/tools/coqrst/coqdomain.py

```diff
@@ -715,6 +715,10 @@ class CoqtopDirective(Directive):
       - ``warn``: Don't die if a command emits a warning
       - ``restart``: Send a ``Restart`` command before running this block (only works in proof mode)
       - ``abort``: Send an ``Abort All`` command after running this block (leaves all pending proofs if any)
+      - ``extra``: if environment variable 'COQRST_EXTRA' is set (to anything else than '0') this is ignored, otherwise behaves as ``fail``
+        This is typically used to show case examples of things outside coq-core or rocq-init.
+        Be careful when using it to surround the code block with a sentence explaining what
+        extra requirement is needed to compile it.
```

Comment:
```suggestion
      - ``extra``: if environment variable 'COQRST_EXTRA' is set (to anything else than '0') this block is ignored, otherwise behaves as ``fail``
        Use this for examples that depend on packages beyond coq-core or rocq-init.
        When you use it, preface the code block with a sentence identifying the extra
        packages needed to run the example.
```

"ignored" meaning it is not passed to coqtop at all?  If it will still appear in the manual, please say that.

This info should be included in `doc/README.md`.  It should also explain how to satisfy such dependencies (OK if refers to a section of the doc covering that).

Also mention that `extra` must appear on each subsequent `.. coqtop ::` that depends on additional packages--until the next `restart` or `abort`.

Reply:
> "ignored" meaning it is not passed to coqtop at all? If it will still appear in the manual, please say that.

No, the block is not ignored. What this says is that `.. coqtop:: extra` behaves as `.. coqtop:: fail`, unless `COQRST_EXTRA` is set (in which case, it's like `extra` was not provided).

Maybe the explanation is too close to the technical aspects and does not sufficiently explain its goals. The logic is that we don't want local builds to fail if you don't have all the dependencies of the examples. But in CI, `COQRST_EXTRA` will be set, so that the examples are properly checked.